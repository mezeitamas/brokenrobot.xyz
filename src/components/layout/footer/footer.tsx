import React from 'react';
import type { FunctionComponent, ReactElement } from 'react';

import { FiRss, FiGithub, FiLinkedin, FiTwitter } from 'react-icons/fi';
import { TbBrandMastodon } from 'react-icons/tb';

import { ExternalLink } from '../../external-link/external-link';

type FooterProps = {
    githubLink: string;
    linkedinLink: string;
    mastodonLink: string;
    twitterLink: string;
    authorName: string;
};

const Footer: FunctionComponent<FooterProps> = ({
    githubLink,
    linkedinLink,
    mastodonLink,
    twitterLink,
    authorName
}): ReactElement => {
    return (
        <footer className="mb-6 mt-16 flex flex-col gap-4">
            <hr />

            <div className="flex flex-col justify-between gap-6 whitespace-nowrap text-gray-700 sm:flex-row sm:gap-10">
                <div className="flex flex-col gap-4 sm:flex-row">
                    <div className="flex flex-row gap-4">
                        <div className="flex items-center gap-2">
                            <FiGithub size="18" />
                            <ExternalLink href={githubLink}>GitHub</ExternalLink>
                        </div>

                        <div className="flex items-center gap-2">
                            <FiLinkedin size="18" />
                            <ExternalLink href={linkedinLink}>LinkedIn</ExternalLink>
                        </div>
                    </div>

                    <div className="flex flex-row gap-4">
                        <div className="flex items-center gap-2">
                            <TbBrandMastodon size="18" />
                            <ExternalLink
                                href={mastodonLink}
                                rel="me"
                            >
                                Mastodon
                            </ExternalLink>
                        </div>

                        <div className="flex items-center gap-2">
                            <FiTwitter size="18" />
                            <ExternalLink href={twitterLink}>Twitter</ExternalLink>
                        </div>
                    </div>
                </div>

                <div className="flex items-center gap-2">
                    <FiRss size="18" />
                    <ExternalLink href="/rss.xml">RSS Feed</ExternalLink>
                </div>
            </div>

            <div className="text-sm text-gray-400">&copy; 2023 {authorName}. All rights reserved.</div>
        </footer>
    );
};

export { Footer };

export type { FooterProps };
