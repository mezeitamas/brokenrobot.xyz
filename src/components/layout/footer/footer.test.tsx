import React from 'react';

import { create } from 'react-test-renderer';

import { Footer } from './footer';
import type { FooterProps } from './footer';

describe('<Footer />', () => {
    test('should render', () => {
        // Given
        const props: FooterProps = {
            githubLink: 'https://github.com/mezeitamas',
            linkedinLink: 'https://www.linkedin.com/in/mezeitamas/',
            mastodonLink: 'https://mastodon.social/@brokenrobot_xyz',
            twitterLink: 'https://twitter.com/brokenrobot_xyz',
            authorName: 'Tamas Mezei'
        };

        // When
        const component = create(<Footer {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
