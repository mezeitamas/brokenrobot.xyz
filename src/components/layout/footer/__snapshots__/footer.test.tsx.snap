// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 1`] = `
<footer
  className="mb-6 mt-16 flex flex-col gap-4"
>
  <hr />
  <div
    className="flex flex-col justify-between gap-4 whitespace-nowrap text-gray-700 sm:flex-row"
  >
    <div
      className="flex gap-6 sm:gap-10"
    >
      <div
        className="flex items-center gap-2"
      >
        <svg
          fill="none"
          height="18"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
          />
        </svg>
        <span
          className="inline-flex items-center gap-1 hover:underline"
        >
          <a
            href="https://github.com/mezeitamas"
            target="_blank"
          >
            GitHub
          </a>
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <polyline
              points="15 3 21 3 21 9"
            />
            <line
              x1="10"
              x2="21"
              y1="14"
              y2="3"
            />
          </svg>
        </span>
      </div>
      <div
        className="flex items-center gap-2"
      >
        <svg
          fill="none"
          height="18"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
          />
          <rect
            height="12"
            width="4"
            x="2"
            y="9"
          />
          <circle
            cx="4"
            cy="4"
            r="2"
          />
        </svg>
        <span
          className="inline-flex items-center gap-1 hover:underline"
        >
          <a
            href="https://www.linkedin.com/in/mezeitamas/"
            target="_blank"
          >
            LinkedIn
          </a>
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <polyline
              points="15 3 21 3 21 9"
            />
            <line
              x1="10"
              x2="21"
              y1="14"
              y2="3"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="flex items-center gap-2"
    >
      <svg
        fill="none"
        height="18"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 11a9 9 0 0 1 9 9"
        />
        <path
          d="M4 4a16 16 0 0 1 16 16"
        />
        <circle
          cx="5"
          cy="19"
          r="1"
        />
      </svg>
      <span
        className="inline-flex items-center gap-1 hover:underline"
      >
        <a
          href="/rss.xml"
          target="_blank"
        >
          RSS Feed
        </a>
        <svg
          fill="none"
          height="14"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <polyline
            points="15 3 21 3 21 9"
          />
          <line
            x1="10"
            x2="21"
            y1="14"
            y2="3"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="text-sm text-gray-400"
  >
    Â© 2023 
    Tamas Mezei
    . All rights reserved.
  </div>
</footer>
`;
